.github/workflows/package-ffi-engine.yml

```diff
@@ -54,12 +61,33 @@ jobs:
             target/
           key: ${{ runner.os }}-${{ matrix.platform.target}}-cargo-${{ hashFiles('**/Cargo.lock') }}
 
-      - name: Build Rust ${{ matrix.platform.name }}
-        uses: actions-rs/cargo@v1
-        with:
-          command: build
-          args: --target ${{ matrix.platform.target }} --release
-          use-cross: ${{ matrix.platform.use_cross }}
+      - name: Install and configure dependencies
+        run: |
+          # dependencies are only needed on ubuntu as that's the only place where
+          # we make cross-compilation
+          if [[ "${{ matrix.platform.os }}" =~ ^ubuntu.*$ ]]; then
+            sudo apt-get install -qq musl-tools musl-dev crossbuild-essential-arm64
+          fi
+
+          # some additional configuration for cross-compilation on linux
+          cat >>~/.cargo/config <<EOF
+          [target.aarch64-unknown-linux-gnu]
+          linker = "aarch64-linux-gnu-gcc"
+          [target.aarch64-unknown-linux-musl]
+          linker = "aarch64-linux-gnu-gcc"
```

Comment:
is it correct?

Reply:
Not sure, it was required to get the engine compiled with openssl support for musl

I got the example from https://github.com/nicolas-van/rust-cross-compile-example/blob/main/.github/workflows/rust.yml#L83-L88

I now realize though that in their example they built a binary, and we are building a library so this may not actually work once its called from the host machine (ie alpine) if it is still linked against glibc

Reply:
AWS uses it in the [docs](https://github.com/awslabs/aws-lambda-web-adapter/blob/5d7885bd26b3b838e0884865f301ab01e14d8960/docs/development.md#compiling-on-macos) like 
```toml
[target.aarch64-unknown-linux-musl] 
linker = "aarch64-unknown-linux-musl-gcc"
```

Reply:
yeah let me try that.

i downloaded the engine built for musl and it does seem like its linked against glibc, which wont work I believe on alpine:

![CleanShot 2024-08-24 at 13 15 57](https://github.com/user-attachments/assets/ee2f7c7c-effa-47a7-b16e-bbf9bb0ceaf4)

Reply:
It cant seem to find `aarch64-linux-musl-gcc` pr ` aarch64-unknown-linux-musl-gcc` (I tried both)

https://github.com/flipt-io/flipt-client-sdks/actions/runs/10540369969/job/29204942815#step:6:305

I think `aarch64-unknown-*` might be specific to MacOS/Homebrew per https://github.com/awslabs/aws-lambda-web-adapter/blob/5d7885bd26b3b838e0884865f301ab01e14d8960/docs/development.md#compiling-on-macos

Also it seems others recommend setting  the linker to `aarch64-linux-gnu-gcc` 🤷🏻 

https://github.com/rust-lang/stacker/issues/80#issuecomment-1547991131

Reply:
Could this warning have an effect on it?
```
warning: openssl-sys@0.9.103: Compiler family detection failed due to error: ToolNotFound: Failed to find tool. Is `aarch64-linux-musl-gcc` installed?
```
I see that someone installed it before the build in the [action](https://github.com/sxyazi/yazi/pull/759/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R25-R27)

Reply:
> yeah let me try that.
> 
> i downloaded the engine built for musl and it does seem like its linked against glibc, which wont work I believe on alpine:
> 
> ![CleanShot 2024-08-24 at 13 15 57](https://private-user-images.githubusercontent.com/209477/361158540-ee2f7c7c-effa-47a7-b16e-bbf9bb0ceaf4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ1MjY5MDIsIm5iZiI6MTcyNDUyNjYwMiwicGF0aCI6Ii8yMDk0NzcvMzYxMTU4NTQwLWVlMmY3YzdjLWVmZmEtNDdhNy1iMTZlLWJiZjliYjBjZWFmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyNFQxOTEwMDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZjM1ZDk2ZjA1NzgwOWU5MWU2NzA1NTYwNjNiY2UxMzZmOTQ3MGYwMWY5NmRhOTZkZGU4Yjc5OWUzOGE2NmQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4QkolwBQZVuHAagRQ_YitgsqgZ0BrUG3uWAAcoJcb-s)

maybe it will work.

Reply:
I switched it back, it builds now again without any warning: https://github.com/flipt-io/flipt-client-sdks/actions/runs/10542101107/job/29208497781

I havent yet tested it in an actual alpine container yet though.. trying to get that setup now

Reply:
It does not like that:

```
 > [6/6] RUN go test ./...:                                                                                                                                             
4.814 FAIL	go.flipt.io/flipt-client [build failed]                                                                                                                 
4.814 FAIL                                                                                                                                                              
4.814 # go.flipt.io/flipt-client.test                                                                                                                                   
4.814 /usr/local/go/pkg/tool/linux_arm64/link: running gcc failed: exit status 1                                                                                        
4.814 /usr/lib/gcc/aarch64-alpine-linux-musl/12.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: warning: libc.so.6, needed by /src/ext/linux_arm64/libfliptengine.so, not found (try using -rpath or -rpath-link)
4.814 /usr/lib/gcc/aarch64-alpine-linux-musl/12.2.1/../../../../aarch64-alpine-linux-musl/bin/ld: warning: ld-linux-aarch64.so.1, needed by /src/ext/linux_arm64/libfliptengine.so, not found (try using -rpath or -rpath-link)
```